src/JustSaying/Fluent/PublishConfig/DynamicAddressMessagePublisher.cs

```diff
@@ -61,6 +61,7 @@ public async Task PublishAsync(Message message, PublishMetadata metadata, Cancel
         var config = _staticConfigBuilder(topicArn);
 
         _ = _publisherCache.TryAdd(topicArn, config.Publisher);
+        lockObj.Release(1);
```

Comment:
I think you need to do this in the batch version here and in the other file too.

Reply:
Good catch, I've added that now